identity/src/main/java/com/android/identity/trustmanagement/CertificateExtensions.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.trustmanagement
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.x500.X500Name
+import org.bouncycastle.asn1.x500.style.BCStyle
+import java.security.cert.X509Certificate
+import javax.security.auth.x500.X500Principal
+
+private const val DIGITAL_SIGNATURE = 0
+private const val KEY_CERT_SIGN = 5
+
+/**
+ * Determine whether the certificate has the key usage of a document signer
+ */
+fun X509Certificate.hasKeyUsageDocumentSigner(): Boolean {
+    if (this.keyUsage == null)
+    {
+        return false
+    }
+    return this.keyUsage[DIGITAL_SIGNATURE]
+}
+
+/**
+ * Determine whether the certificate has the key usage of a certificate authority
+ */
+fun X509Certificate.hasKeyUsageCaCertificate(): Boolean {
+    if (this.keyUsage == null)
+    {
+        return false
+    }
+    return this.keyUsage[KEY_CERT_SIGN]
+}
+
+/**
+ * Extract the common name of a X500Principal (subject or issuer)
+ */
+fun X500Principal.getCommonName(defaultValue: String): String {
+    return readRdn(this.name, BCStyle.CN, defaultValue)
+}
+
+/**
+ * Extract the organisation of a X500Principal (subject or issuer)
+ */
+fun X500Principal.getOrganisation(defaultValue: String): String {
+    return readRdn(this.name, BCStyle.O, defaultValue)
+}
+
+/**
+ * Extract the organisational unit of a X500Principal (subject or issuer)
+ */
+fun X500Principal.organisationalUnit(defaultValue: String): String {
+    return readRdn(this.name, BCStyle.OU, defaultValue)
+}
+
+/**
+ * Extract the country code of a X500Principal (subject or issuer)
+ */
+fun X500Principal.countryCode(defaultValue: String): String {
+    return readRdn(this.name, BCStyle.C, defaultValue)
+}
+
+/**
+ * Read a relative distinguished name from a distinguished name
+ */
+private fun readRdn(name: String, field: ASN1ObjectIdentifier, defaultValue: String): String {
+    val x500name = X500Name(name)
+    for (rdn in x500name.getRDNs(field)) {
+        val attributes = rdn.typesAndValues
+        for (attribute in attributes) {
+            return attribute.value.toString()
+        }
+    }
+    return defaultValue
+}
```

Comment:
This all looks like utility functions. If so, move to appropriate `TrustManagerUtil.kt` and make it all private.

Reply:
Those functions are implemented as extensions, so that they can be reused, like `certificate.subjectX500Principal.getCommonName("")`.

In the class ShowDocumentFragment.kt this was done this way:

           `val issuerItems = certChain.last().issuerX500Principal.name.split(",")
            var cnFound = false
            val commonName = StringBuffer()
            for (issuerItem in issuerItems) {
                when {
                    issuerItem.contains("CN=") -> {
                        val (key, value) = issuerItem.split("=", limit = 2)
                        commonName.append(value)
                        cnFound = true
                    }
                    // Common Name value with ',' symbols would be treated as set of items
                    // Append all parts of CN field if any before next issuer item
                    cnFound && !issuerItem.contains("=") -> commonName.append(", $issuerItem")
                    // Ignore any next issuer items only after we've collected required
                    cnFound -> break
                }`
Now it is only one line of code...